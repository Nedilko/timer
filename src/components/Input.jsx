import {useState} from 'react'
import PropTypes from 'prop-types'

Input.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string
}

function Input({label, placeholder = 'Greetings text'}) {
  const [value, setValue] = useState('')
  const handleChange = () => {
    setValue(event.target.value)
  }

  return (
    <label className='py-2'>{label}
      <input onChange={handleChange} className='mx-1 px-1 rounded bg-gray-200' type="text" placeholder={placeholder}/>
    </label>
  )
}

export default Input