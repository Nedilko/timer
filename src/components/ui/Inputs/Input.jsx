import { useEffect, useState } from "react";
import PropTypes from "prop-types";

Input.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

function Input({ value, label, placeholder = "Greetings text", onChange }) {
  const [textValue, setTextValue] = useState(value);
  const handleChange = (e) => {
    setTextValue(e.target.value);
  };

  useEffect(() => {
    onChange(textValue);
  }, [textValue, onChange]);

  return (
    <label role="label" className="py-2 select-none">
      {label}
      <input
        onChange={handleChange}
        className="mx-1 px-1 rounded bg-gray-200"
        type="text"
        placeholder={placeholder}
        value={textValue}
      />
    </label>
  );
}

export default Input;
