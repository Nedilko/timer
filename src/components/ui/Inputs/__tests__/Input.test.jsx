import Input from "../Input";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Input", () => {
  const clickHandler = jest.fn();
  beforeEach(() => {
    clickHandler.mockClear();
  });

  it("should render", () => {
    render(
      <Input
        label="test label"
        value="test value"
        placeholder="test holder"
        onChange={clickHandler}
      />
    );

    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  it("should show label", () => {
    render(
      <Input
        label="test label"
        value="test value"
        placeholder="test holder"
        onChange={clickHandler}
      />
    );

    expect(screen.getByLabelText("test label")).toBeInTheDocument();
  });

  it("should show value", () => {
    render(
      <Input
        label="test label"
        value="test value"
        placeholder="test holder"
        onChange={clickHandler}
      />
    );

    expect(screen.getByDisplayValue("test value")).toBeInTheDocument();
  });

  it("should show placeholder", () => {
    render(
      <Input
        label="test label"
        value="test value"
        placeholder="test holder"
        onChange={clickHandler}
      />
    );

    expect(screen.getByPlaceholderText("test holder")).toBeInTheDocument();
  });

  it("should handle onChange", () => {
    render(
      <Input
        label="test label"
        value="test value"
        placeholder="test holder"
        onChange={clickHandler}
      />
    );

    userEvent.type(screen.getByRole("textbox"), "test");

    expect(clickHandler).toBeCalledTimes(5);
  });
});
