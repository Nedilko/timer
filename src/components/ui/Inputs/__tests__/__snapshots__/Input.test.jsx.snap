// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<input
  class="mx-1 px-1 rounded bg-gray-200"
  placeholder="test holder"
  type="text"
  value="test value"
/>
`;
